specs/commandline/plugin.md

```diff
@@ -41,51 +41,115 @@ Aliases:
   -h, --help                    help for install
   -f, --force                   force the installation of a plugin
+      --sha256sum string        must match SHA256 of the plugin source
+      --url                     install a plugin from a remote HTTPS address
 
 Aliases:
   install, add
 ```
 
-### notation plugin remove
+### notation plugin uninstall
 
 ```text
-Removes a plugin
+Uninstall a plugin
 
 Usage:
-  notation plugin remove [flags] <plugin name>
+  notation plugin uninstall [flags] <plugin_name>
 
 Flags:
   -h, --help          help for remove
-
+  -y, --yes           do not prompt for confirmation
 Aliases:
-  remove, rm, uninstall, delete
+  uninstall, remove, rm
 ```
 
 ## Usage
 
-### Install a plugin
+## Install a plugin 
+
+### Install a plugin from file system
+
+Install a Notation plugin from file system. Plugin file supports `.zip` and `.tar.gz` format. The checksum validation is optional for this case. 
 
 ```shell
-notation plugin install <plugin path>
+$ notation plugin install --file <file_path>
+```
+
+Upon successful execution, the plugin is copied to Notation's plugin directory. If the plugin directory does not exist, it will be created. The name and version of the installed plugin are displayed as follows. 
+
+```console
+Successfully installed plugin <plugin name>, version <x.y.z>
+```
+
+If the entered plugin checksum digest doesn't match the published checksum, Notation will return an error message and will not start installation.
+
+```console
+Error: failed to install the plugin, input checksum does not match the published checksum, expected <digest>
+```
+
+If the plugin version is higher than the existing plugin, Notation will start installation and overwrite the existing plugin.
+
+```console
+Successfully installed plugin <plugin name>, updated the version from <x.y.z> to <a.b.c>
+```
+
+If the plugin version is equal to the existing plugin, Notation will not start installation and return the following message. Users can use a flag `--force` to skip plugin version check and force the installation.
+
+```console
+Plugin <plugin name>-<x.y.z> already exists. 
+To view a list of installed plugins, use "notation plugin list".
+If you want to install the same version, use "notation plugin uninstall" to uninstall it before re-installing the same version or use "--force" to force the installation
+```
+
+If the plugin version is lower than the existing plugin, Notation will return an error message and will not start installation. Users can use a flag `--force` to skip plugin version check and force the installation.
+
+```console
+Error: failed to install the plugin, plugin version is lower than the existing plugin. 
+To view a list of installed plugins, use "notation plugin list".
+If you want to install an old version, use "notation plugin uninstall" to uninstall the existing plugin before installation or use "--force" to force the installation
```

Comment:
same here,
```suggestion
```

Reply:
For older version, it is worth mentioning, but it's better we highlights not recommended, but in rare case, if users want to do that.

Reply:
Installing an older version is an downgrade operation. Users may consider downgrade for some reasons. I agree with @yizha1 that mentioned it but state that installing an older version is not recommended.

Reply:
Updated